AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -918,13 +930,88 @@ public class AWSAppSyncClient {
         }
     }
     
+    /// Performs a sync operation where a base query is periodically called to fetch primary data from the server based on the syncConfiguration.
+    ///
+    /// - Parameters:
+    ///   - baseQuery: The base query to fetch which contains the primary data.
+    ///   - baseQueryResultHandler: Closure that is called when base query results are available or when an error occurs. Every time a sync operation is called, a fetch for the baseQuery from the cache will be done first before initiating any other operations.
+    ///   - deltaQuery: The delta query which fetches data starting from the `lastSync` time. Defaults to a no-op query.
+    ///   - deltaQueryResultHandler: Closure that is called when delta query executes. Defaults to a no-op handler.
+    ///   - callbackQueue: An optional queue on which sync callbacks will be invoked. Defaults to the main queue.
+    ///   - subscriptionWithSyncConfiguration: The sync configuration where the baseQuery sync interval can be specified. (Defaults to 24 hours.)
+    /// - Returns: An object that can be used to cancel the sync operation.
+    public func sync<BaseQuery: GraphQLQuery, DeltaQuery: GraphQLQuery>(baseQuery: BaseQuery,
+                                                                        baseQueryResultHandler: @escaping OperationResultHandler<BaseQuery>,
+                                                                        deltaQuery: DeltaQuery? = nil,
+                                                                        deltaQueryResultHandler: DeltaQueryResultHandler<DeltaQuery>? = nil,
+                                                                        callbackQueue: DispatchQueue = DispatchQueue.main,
+                                                                        subscriptionWithSyncConfiguration: SubscriptionWithSyncConfiguration? = nil) -> Cancellable {
```

Comment:
can we rather use default parameter of SubscriptionWithSyncConfiguration()?

```swift
subscriptionWithSyncConfiguration:  SubscriptionWithSyncConfiguration = SubscriptionWithSyncConfiguration()
```

Reply:
I think default value that is not `nil` is clearer. `nil` may be more memory effective, but sync config is created in the function body below. So it doesn't matter in this case.

Reply:
Also, you can't use non-statically-created values as defaults--that was my first thought too :)

Reply:
Revised and included `SyncConfiguration()` as the default here and throughout. I had mistakenly thought that using an instance constructor wasn't allowed for defaults; the problem was actually that the constructor was `internal` rather than `public`, as @rohandubal noted elsewhere.